azurerm/internal/services/subscription/subscription_resource.go

```diff
@@ -0,0 +1,500 @@
+	subscriptionsClient := meta.(*clients.Client).Subscription.Client
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SubscriptionAliasID(d.Id())
+	if err != nil {
+		return err
+	}
+	d.Set("alias", id.Name)
+
+	alias, err := client.Get(ctx, id.Name)
+	if err != nil {
+		return fmt.Errorf("reading Subscription Aliss %q: %+v", id.Name, err)
+	}
+
+	subscriptionId := ""
+	subscriptionName := ""
+	tenantId := ""
+	if props := alias.Properties; props != nil && props.SubscriptionID != nil {
+		subscriptionId = *props.SubscriptionID
+		resp, err := subscriptionsClient.Get(ctx, subscriptionId)
+
+		if err != nil {
+			return fmt.Errorf("failed to read Subscription %q (Alias %q) for Tenant Information: %+v", subscriptionId, id.Name, err)
+		}
+		if resp.TenantID != nil {
+			tenantId = *resp.TenantID
+		}
+
+		if resp.DisplayName != nil {
+			subscriptionName = *resp.DisplayName
+		}
+
+		state := ""
+		if resp.State == subscriptions.Disabled || resp.State == subscriptions.Warned {
+			state = "Cancelled"
+		}
+
+		if resp.State == subscriptions.Enabled {
+			state = "Active"
+		}
+
+		d.Set("state", state)
+	}
+
+	// (@jackofallops) A subscription's billing scope is not exposed in any way in the API/SDK so we cannot read it back here
+
+	d.Set("subscription_id", subscriptionId)
+	d.Set("subscription_name", subscriptionName)
+	d.Set("tenant_id", tenantId)
+
+	return nil
+}
+
+// (@jackofallops) - Delete here is a misnomer.  The nature of subscriptions is such that they are never truly deleted
+// Deleted here means Cancelled. Cancelled subscriptions are held in this state for  90 days before being purged from
+// active systems.  However, the backend billing systems _never_ remove this data, so once a Subscription ID has been
+// used and purged from active use it can never be recovered nor the UUID reused.
+// Note Cancelling a Subscription leaves it in one of several states, `Disabled` for a Subscription with no Resources or
+// Alias assignments, `Warned` for Cancelled with "something" associated with it.
+func resourceSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Subscription.AliasClient
+	subscriptionClient := meta.(*clients.Client).Subscription.SubscriptionClient
+	subscriptionsClient := meta.(*clients.Client).Subscription.Client
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SubscriptionAliasID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Get subscription details for later
+	alias, err := client.Get(ctx, id.Name)
+	if err != nil || alias.Properties == nil {
+		return fmt.Errorf("could not read Alias %q for Subscription: %+v", id.Name, err)
+	}
+	subscriptionId := ""
+	if subscriptionIdRaw := alias.Properties.SubscriptionID; subscriptionIdRaw != nil {
+		subscriptionId = *subscriptionIdRaw
+	}
+
+	sub, err := subscriptionsClient.Get(ctx, subscriptionId)
```

Comment:
similar to #1608 - should we conditionally (feature-flag, defaulted off) check for the presence of existing resources prior to cancelling this?

Reply:
As discussed offline, despite the docs suggesting that Subscriptions can be cancelled regardless of resources, the API rejects cancellations where resources are present in the Subscription. PR description and docs updated to reflect this.